File: c++/Valid Anagram.cpp
#include<iostream>
#include<string>

class Solution {
public:
    bool isAnagram(std::string s, std::string t) {
        if (s.size() != t.size()) return false;

        int count[26] = {0};

        for (size_t i = 0; i < s.size(); i++) {
            count[s[i] - 'a']++;
            count[t[i] - 'a']--;
        }

        for (int i = 0; i < 26; i++) {
            if (count[i] != 0) return false;
        }

        return true;  
    }
};

// Test cases
int main() {
   Solution solution;
   
   // case 1
   std::string s1 = "anagram";
   std::string t1 = "nagaram";
   std::cout << "result of test case 1 is : " << solution.isAnagram(s1, t1) << std::endl; // Expected: true
   // case 2
   std::string s2 = "rat";
   std::string t2 = "car";
   std::cout << "result of test case 2 is : " << solution.isAnagram(s2, t2) << std::endl; // Expected: false
   
   return 0;
}

File: python/Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        count = [0] * 26

        for ch1, ch2 in zip(s, t):
            count[ord(ch1) - ord('a')] += 1
            count[ord(ch2) - ord('a')] -= 1

        return all(c == 0 for c in count)
# Test cases
if __name__ == "__main__":
    solution = Solution();

    # case 1
    s1 = "anagram"
    t1 = "nagaram"
    print(f"result of test case 1 : {solution.isAnagram(s1, t1)}") # Expected: true
    
    # case 2
    s2 = "rat"
    t2 = "car"
    print(f"result of test case 2 : {solution.isAnagram(s2, t2)}") # Expected: false
